pkg/test/assertions.go

```diff
@@ -1,18 +1,16 @@
-package assertions
+package test
```

Comment:
We could also add `UpdateResource` in order to clean up our duplicated update functions on toolchain-e2e repo (ex. UpdateUserSignup, UpdateSpaceBinding, etc.). Something like this:

```
func UpdateResource[T any](
    t T,
    client client.Client,
    namespace string,
    resourceName string,
    status bool,
    modifyResource func(resource *T),
) (*T, error) {
    var resource *T
    err := wait.PollUntilContextTimeout(context.TODO(), time.Second*2, time.Minute*5, true, func(ctx context.Context) (done bool, err error) {
        // Fetch the resource from the cluster
        freshResource := new(T)
        if err := client.Get(context.TODO(), types.NamespacedName{Namespace: namespace, Name: resourceName}, freshResource); err != nil {
            return true, err
        }

        // Get the resource type name
        resourceType := reflect.TypeOf(freshResource).Elem().Name()

        // Apply the modification function
        modifyResource(freshResource)

        // Update resource Spec or Status
        if status {
            if err := client.Status().Update(context.TODO(), freshResource); err != nil {
                t.Logf("Error updating %s Status '%s': %s. Will retry...", resourceType, resourceName, err.Error())
                return false, nil
            }
        } else {
            if err := client.Update(context.TODO(), freshResource); err != nil {
                t.Logf("Error updating %s Spec '%s': %s. Will retry...", resourceType, resourceName, err.Error())
                return false, nil
            }
        }

        resource = freshResource
        return true, nil
    })

    return resource, err
}
```